packages/app/obojobo-repository/shared/components/collection-manage-modules.js

```diff
@@ -0,0 +1,103 @@
+require('./collection-manage-modules-dialog.scss')
+
+const React = require('react')
+const { useEffect, useState } = require('react')
+const ReactModal = require('react-modal')
+const Button = require('./button')
+const ModuleSearchDialog = require('./module-search-dialog-hoc')
+const CollectionImage = require('./collection-image')
+const CollectionModuleListItem = require('./collection-module-list-item')
+
+const CollectionManageModulesDialog = props => {
+	const [modulePickerOpen, setModulePickerOpen] = useState(false)
+
+	// Load the modules in this collection on the initial render
+	useEffect(() => {
+		props.loadCollectionModules(props.collection.id)
+	}, [])
+
+	const openModulePicker = () => {
+		setModulePickerOpen(true)
+	}
+
+	const closeModulePicker = () => {
+		setModulePickerOpen(false)
+	}
+
+	const addModule = draftId => {
+		props.collectionAddModule(draftId, props.collection.id)
+		closeModulePicker()
+	}
+
+	const removeModule = draftId => {
+		props.collectionRemoveModule(draftId, props.collection.id)
+	}
+
+	let searchModalRender = null
+
+	if (modulePickerOpen) {
+		searchModalRender = (
+			<ReactModal
+				isOpen={true}
+				contentLabel="Module Search"
+				className="repository--modal"
+				overlayClassName="repository--modal-overlay"
+				onRequestClose={closeModulePicker}
+			>
+				<ModuleSearchDialog
+					collectionId={props.collection.id}
+					collectionModules={props.collectionModules}
+					onClose={closeModulePicker}
+					onSelectModule={addModule}
+				/>
+			</ReactModal>
+		)
+	}
+
+	let modulesRender = null
+
+	if (props.collectionModules) {
+		modulesRender = props.collectionModules.map(module => (
+			<CollectionModuleListItem key={module.draftId} {...module}>
+				<Button
+					className="close-button"
+					onClick={() => {
+						removeModule(module.draftId)
+					}}
+				>
+					×
+				</Button>
+			</CollectionModuleListItem>
+		))
+	}
+
+	return (
+		<div className="collection-manage-modules-dialog">
+			<div className="top-bar">
+				<CollectionImage id={props.collection.id} />
+				<div className="collection-title">{props.collection.title}</div>
+				<Button className="close-button" onClick={props.onClose}>
```

Comment:
These need an aria-label of "close", I forget if this Button component allows that